docs/cross-platform.md

```diff
@@ -0,0 +1,119 @@
+
+The good news is that the majority of OCaml build tools do support building from a `cygwin` environment - and often, that is the recommended environment for building on Windows.
+
+Note that, even though our build commands on Windows are executed in a `cygwin` shell, the __binaries that are produced are truly native__. This applies both to all the packages we build, as well as the resultant binary for your project itself. Cygwin provides two compilers - a cygwin `gcc` compiler, and a `mingw` cross-compiler that compiles to native windows executables. We leverage the latter so the result is that `esy build` on Windows gives you native executables. This means that you can use `esy` to build apps natively for any platform!
+
+## Platform Differences
+
+### Path considerations
+
+A major difference with Windows vs OSX/Linux is the path separator - `\` vs `/`. Windows can actually work with _either_ path separator (for the most part). 
+
+The challenge, though, is `\` is also the __escape character__ - so we can hit double-escaping issues when we render the strings.
+
+For the time being, we'll __normalize all paths to use a forward slash__ - this keeps us from hitting the 'double-escaping issues' with backslash.
+
+One other complexity is that we use Cygwin for running our build tasks, and the canonical paths for Cygwin are POSIX style, like:
+- `C:\temp` -> `/cygdrive/c/temp`
+
+The majority of Cygwin utilities can work with either style of paths - therefore, we will prefer the __Windows path, normalized with a forward slash__, for shell invocations.
+
+Some utilities that Cygwin ships with do not support any style of Windows path (either forward or back slash). For those utilities - like `rsync` - we will need to convert the path to a cygwin-style path. Cygwin includes the `cygpath` utility for this, which can convert paths of the form `C:/temp` or `C:\\temp` to cygwin-style paths like `/cygdrive/c/temp`. These exceptions should be few, and primarily encountered in bootstrapping the `esy` environment internally, as opposed to present in the build scripts.
+
+### Cross-platform environment variables
+
+Another difference in Windows vs OSX/Linux is how the environment variables separate individual paths. On POSIX systems, it's a colon - `:` - on Windows, its `;`
+
+Because our __built artifacts are native__, we must use native Windows paths in our environment variables (normalized with forward slash).
+
+### Symlinks
+
+Symlinks on Windows are not well-supported today. To create symlinks, a Windows user either needs to be running as _administrator_, or set up a special developer mode - neither which is the default behavior on the system.
+
+Therefore, at this time, we should avoid using symlinks in cross-platform code.
```

Comment:
We could note that esy currently uses symlinks in one place, and for the time being you must run in an administrator shell (but that eventually that limitation should go away).

Reply:
Discussing with @andreypopp - I believe the plan is to remove symlinks completely from `esy`. So it might not even be worth bringing it up, if that change is in the near-term.